Classes/Emogrifier.php

```diff
@@ -671,6 +671,7 @@ private function copyCssWithMediaToStyleNode(\DOMDocument $xmlDocument, \DOMXPat
             foreach ($this->parseSelectors($mediaQuery['css']) as $selector) {
                 if ($this->existsMatchForCssSelector($xpath, $selector['selector'])) {
                     $mediaQueriesRelevantForDocument[] = $mediaQuery['query'];
+                    break;
```

Comment:
The 2 issues are related and I think it's not a good idea to complicate things with separate PRs. I have modified the unit tests so that they fail if a selector is added more than once.

So I strongly suggest to keep them in a single PR, to keep things simple, but you're the boss.